mode/verilog/verilog.js

```diff
@@ -354,11 +360,189 @@ CodeMirror.defineMode("verilog", function(config, parserConfig) {
   };
 });
 
-CodeMirror.defineMIME("text/x-verilog", {
-  name: "verilog"
-});
-CodeMirror.defineMIME("text/x-systemverilog", {
-  name: "systemverilog"
+  CodeMirror.defineMIME("text/x-verilog", {
+    name: "verilog"
+  });
+
+  CodeMirror.defineMIME("text/x-systemverilog", {
+    name: "verilog"
+  });
+
+  CodeMirror.defineMIME("text/x-svx", {
+    name: "verilog",
+    vxModeIsOn: true,
+    vxHooks: {
+      "\\": function(stream, state, thisMode) {
+        var codeRegion = "", vxIndent = 0, style = false;
+        var curPunc  = stream.string;
+        if ((stream.sol()) && (/\\SV/.test(stream.string))) {
+          curPunc = (/\\SVX_version/.test(stream.string))
+            ? "\\SVX_version" : stream.string;
+          stream.skipToEnd();
+          if (curPunc=="\\SV" && state.vxContext.vxCodeActive) {codeRegion = "2v";};
+          if (/\\SVX/.test(curPunc) && !state.vxContext.vxCodeActive) {codeRegion ="2vx";};
+          if (curPunc=="\\SVX_version" && state.vxContext.vxCodeActive) {codeRegion = "2vx";};
+          style=(thisMode.svxkeywords.propertyIsEnumerable(curPunc))
+            ? thisMode.svxkeywords[curPunc] : "builtin";
+          state.vxContext.svxCurCtlFlowChar  = state.vxContext.svxPrevPrevCtlFlowChar
+            = state.vxContext.svxPrevCtlFlowChar = "";
+          if (codeRegion == "2vx") {
+            state.vxContext.svxCurCtlFlowChar  = "\\";
+            vxIndent=thisMode.vxHooks["svxGenIndent"](stream, state, thisMode);
+          }
+          if ("2v"  == codeRegion)  {state.vxContext.vxCodeActive=false;}
+          if ("2vx" == codeRegion)  {state.vxContext.vxCodeActive=true;}
```

Comment:
Why indirect through `codeRegion`, rather than simply doing this right away in the `if` statements above?